v2/src/app/services/ordering.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Injectable } from '@angular/core';
+import {HttpClient} from "@angular/common/http";
+import { map } from 'rxjs/operators';
+import {City} from "../interfaces/city";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class OrderingService {
+    private getAllCitiesUrl = '/api/getallcities';
+    private getPlacesUrl = '/api/getplaces';
+    private getSpaceUrl = '/api/getspaces';
+
+    constructor(
+        private http: HttpClient
+    ) { }
+
+
+    getAllCity() {
+        return this.http.get(this.getAllCitiesUrl)
+            .pipe(map((city: any): City =>{
+
+                return city.map(res => {
+                    return {
+                        id: res.id,
+                        name: res.name
+                    }
+                })
+        }));
+    }
+
+    getPlaces(id:number){
+        if (id != 0 ){
+        console.log('getPlaces got not null id params')
```

Comment:
In which case we can get id = 0?

Reply:
wait for the building block. This consolm is intended for the debug. I'll fix it with the backend v2